src/template.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Parse format string and return object
+ *
+ * @param {String} format format string
+ */
+export default function (format) {
+  const regex = /\{\{(\w+)\}\}/g;
+  const result = [];
+
+  if (!format || (typeof format !== 'string' &&  !(format instanceof String))) {
+    return new Function('data', 'return \'\';');
+  }
+
+  let index = 0;
+  let length = 0;
+  let match;
+  while (match = regex.exec(format)) {
+    if (index + length < match.index) {
+      result.push('\'' + format.substring(index + length, match.index) + '\'');
+    }
+    //fast check for a function: !!(object && object.constructor && object.call && object.apply) 
+    result.push('((!!(data.' + match[1] + ' && data.' + match[1] + '.constructor && data.' + match[1] + '.call && data.' + match[1] + '.apply) ? data.' + match[1] + '() : data.' + match[1] + ') || \'' + match[0] + '\')');

```

Comment:
instead of determining whether it's a function at run-time, can you pass the `meta` into the template function and determine whether it's a function by using the value type?

Reply:
I can but it's insignificant change, this check is really fast.

Reply:
Done!